contracts/common/StorageAccessible.sol

```diff
@@ -39,13 +40,15 @@ abstract contract StorageAccessible {
      */
     function simulateAndRevert(address targetContract, bytes memory calldataPayload) external {
         // solhint-disable-next-line no-inline-assembly
+        /// @solidity memory-safe-assembly
         assembly {
             let success := delegatecall(gas(), targetContract, add(calldataPayload, 0x20), mload(calldataPayload), 0, 0)
 
-            mstore(0x00, success)
-            mstore(0x20, returndatasize())
-            returndatacopy(0x40, 0, returndatasize())
-            revert(0, add(returndatasize(), 0x40))
+            let freeMemoryPtr := mload(0x40)
```

Comment:
Could we align the variable names withe the other section where we have this code?

Reply:
Adjusted